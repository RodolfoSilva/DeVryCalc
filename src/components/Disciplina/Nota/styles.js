import { StyleSheet } from 'react-native'
import { colors } from '../../../styles'

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center'
  },
  value: {
    fontSize: 38,
    fontWeight: '200',
    color: colors.mutted
  },
  title: {
    fontSize: 14,
    fontWeight: '300',
    color: colors.mutted
  }
})
