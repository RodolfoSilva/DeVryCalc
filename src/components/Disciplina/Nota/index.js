import React from 'react'
import { Text, View } from 'react-native'
import PropTypes from 'prop-types'
import styles from './styles'

const Nota = ({ value = null, title, style, valueStyle, titleStyle }) => (
  <View style={[styles.container, style]}>
    <Text style={[styles.value, valueStyle]}>{value === null ? '_._' : value}</Text>
    <Text style={[styles.title, titleStyle]}>{title}</Text>
  </View>
)

Nota.propTypes = {
  value: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  style: PropTypes.object,
  valueStyle: PropTypes.func,
  titleStyle: PropTypes.func
}

export default Nota
