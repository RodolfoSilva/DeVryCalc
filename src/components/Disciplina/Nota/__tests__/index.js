import 'react-native'
import React from 'react'
import Nota from '../'
import { shallow } from 'enzyme'

describe('Card', () => {
  it('Renderiza o titulo e no lugar da nota coloca _._', () => {
    const wrapper = shallow(<Nota title="AP1" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renderiza a nota do usuÃ¡rio', () => {
    const wrapper = shallow(<Nota title="AP1" value={5} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Renderiza o titulo e se a nota for null no lugar dela coloca _._', () => {
    const wrapper = shallow(<Nota title="AP1" value={null} />)
    expect(wrapper).toMatchSnapshot()
  })
})
