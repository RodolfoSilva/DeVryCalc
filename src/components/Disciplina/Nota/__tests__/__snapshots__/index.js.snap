// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nota Renderiza a nota do usuário 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#AAA",
          "fontSize": 38,
          "fontWeight": "200",
        },
        undefined,
      ]
    }
  >
    5
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#AAA",
          "fontSize": 14,
          "fontWeight": "300",
        },
        undefined,
      ]
    }
  >
    Média
  </Text>
</View>
`;

exports[`Nota Renderiza o titulo e no lugar da nota coloca _._ 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#AAA",
          "fontSize": 38,
          "fontWeight": "200",
        },
        undefined,
      ]
    }
  >
    _._
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#AAA",
          "fontSize": 14,
          "fontWeight": "300",
        },
        undefined,
      ]
    }
  >
    AP1
  </Text>
</View>
`;

exports[`Nota Renderiza o titulo e se a nota for null no lugar dela coloca _._ 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#AAA",
          "fontSize": 38,
          "fontWeight": "200",
        },
        undefined,
      ]
    }
  >
    _._
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#AAA",
          "fontSize": 14,
          "fontWeight": "300",
        },
        undefined,
      ]
    }
  >
    Title
  </Text>
</View>
`;
